crates/router/src/core/payment_methods/cards.rs

```diff
@@ -627,7 +627,7 @@ pub async fn mock_add_card(
         card_fingerprint: response.card_fingerprint.into(),
         card_global_fingerprint: response.card_global_fingerprint.into(),
         merchant_id: Some(response.merchant_id),
-        card_number: Some(response.card_number.into()),
+        card_number: Some(response.card_number.try_into().unwrap()),
```

Comment:
You can throw this value error 
https://github.com/juspay/hyperswitch/blob/f624eb52d61561c365cce21e58b08281d096d904/crates/router/src/core/errors.rs#L307

Reply:
Add an `attach_printable` to provide clear context as to what failed here. In this case the printable message would be `Invalid card number format from the mock locker`.

Reply:
Like this?
```rust
card_number: { match response.card_number.try_into() {                                                                                                                
               Ok(ccnum) => Some(ccnum),                                                                                       
               Err(_) => return Err(errors::VaultError::ResponseDeserializationFailed).into_report().attach_printable("Invalid card number format from the mock locker")
            }},
```
I have not yet dealt with the `error_stack` crate which I think is where the `attach_printable` method comes from, correct? 
I can also implement `From<CCValError>` for `VaultError` but I felt like this would be kind of displaced. As there are no other `From` implementations for `VaultError`

Reply:
You can do something like this. This would create a `Report` which contains both the initial error (`CCValError` in this case) and the error we'd like to raise (`VaultError::ResponseDeserializationFailed`) from this function.

```rust
card_number: match response.card_number.try_into() {                                                                                                                
                Ok(ccnum) => Some(ccnum),                                                                                       
                error => return error.into_report()
                                .change_context(errors::VaultError::ResponseDeserializationFailed)
                                .attach_printable("Invalid card number format from the mock locker")
            },
```